### DISCUSSION TITLE
Can Kong return JSON body in camel case

### DISCUSSION THREAD

**Discussion by luozhouyang** [Kong/kong#9469] [Help] [answered]
I installed OAuth2 plugin for my service, but the response JSON body was in snake case. Is it possible to convert this snake case JSON to camel case?

**Comment by hanshuebner** [comment, marked as answer]
Hello @luozhouyang,

if you are talking about the token related JSON responses generated by Kong, then no, they cannot be switched to camel case.  Kong implements OAuth2 according to [RFC6749](https://www.rfc-editor.org/rfc/rfc6749) which specifies specific keys to be used in JSON responses, and they happen to be in snake case.

You may be able to implement a plugin that converts the keys for you or try to use the [response-transformer](https://docs.konghq.com/hub/kong-inc/response-transformer/) plugin, but the latter may be tricky as key renaming is not directly supported.

Kind regards,
Hans
